pkg/controllers/provisioning/binpacking/packer.go

```diff
@@ -170,34 +170,19 @@ func (p *Packer) packWithLargestPod(unpackedPods []*v1.Pod, packables []*Packabl
 		return &Packing{Pods: [][]*v1.Pod{bestPackedPods}, InstanceTypeOptions: bestInstances}, remainingPods
 	}
 
-	for i, packable := range packables {
+	for _, packable := range packables {
 		// check how many pods we can fit with the available capacity
 		if result := packable.Pack(unpackedPods); len(result.packed) == maxPodsPacked {
-			// Add all packable nodes that have more resources than this one
-			// Trim the bestInstances so that provisioning APIs in cloud providers are not overwhelmed by the number of instance type options
-			// For example, the AWS EC2 Fleet API only allows the request to be 145kb which equates to about 130 instance type options.
-			for j := i; j < len(packables) && j-i < MaxInstanceTypes; j++ {
-				// packable nodes are sorted lexicographically according to the order of [CPU, memory]
-				// It may result in cases where an instance type may have larger index value when it has more CPU but fewer memory
-				// Need to exclude instance type with smaller memory and fewer pods
-				lv := packables[i]
-				rv := packables[j]
-
-				lvMem := lv.Resources()[v1.ResourceMemory]
-				rvMem := rv.Resources()[v1.ResourceMemory]
-
-				lvPods := lv.Resources()[v1.ResourcePods]
-				rvPods := rv.Resources()[v1.ResourcePods]
-
-				if lvMem.Cmp(rvMem) <= 0 && lvPods.Cmp(rvPods) <= 0 {
-					bestInstances = append(bestInstances, packables[j])
-				}
-			}
+			bestInstances = append(bestInstances, packable)
 			bestPackedPods = result.packed
 			remainingPods = result.unpacked
-			break
+			// reached the max that we care about
+			if len(bestInstances) == MaxInstanceTypes {
```

Comment:
since we filter out instance types in the CP now, it might make sense to just move the MaxInstanceTypes to the CP or increase this value.   It might make the bin-packer do a lot more work though...

Reply:
Alternatively, build this on a branch and whole hog move towards a unified scheduler/binpacker.

Reply:
That's the next step, but right now it sometimes creates instance types that are too small. Was hoping to get this in as a patch before any release.

Reply:
> since we filter out instance types in the CP now, it might make sense to just move the MaxInstanceTypes to the CP or increase this value.   It might make the bin-packer do a lot more work though...  

I think it will with a converged scheduler/bin packer. I think scheduling costs will dominate any bin packing cost anyway.